manage_issues.py

```diff
@@ -0,0 +1,579 @@
+        update_value = get_update_value(
+            project, issue_type, field_name, field_value, True
+        )
+        if update_value:
+            create_fields.update(update_value)
+        else:
+            update_fields[field_name] = field_value  # candidate for update
+    issue = jira.create_issue(fields=create_fields)
+    if update_fields:
+        issue = __update_jira_issue(issue, update_fields)
+    return issue
+
+
+def __github_to_args(ctx):
+    """Update fields in args with given github_url."""
+    ary = ctx["github_url"].split("/")
+    ctx["github_url_api"] = (
+        "https://api.github.com/repos/" + ary[3] + "/" + ary[4] + "/issues/" + ary[-1]
+    )
+    ctx["github_url_api_pr_merge"] = (
+        "https://api.github.com/repos/"
+        + ary[3]
+        + "/"
+        + ary[4]
+        + "/pulls/"
+        + ary[-1]
+        + "/merge"
+    )
+    gh_issue = get_github_issue_pr(ctx)
+    if not ctx["summary"]:
+        ctx["summary"] = gh_issue["title"]
+    if not ctx["description"]:
+        ctx["description"] = gh_issue["body"].replace("\r", "")
+    if not ctx["issue_type"]:
+        if ctx["summary"].startswith("feat: "):
+            ctx["issue_type"] = "Story"
+        elif ctx["summary"].startswith("fix: "):
+            ctx["issue_type"] = "Bug"
+        else:
+            ctx["issue_type"] = "Bug"  # user should give ctx["issue_type"] in this case
+    if not ctx["role"]:
+        ary = ctx["github_url"].split("/")
+        if ary[3] == "linux-system-roles":
+            ctx["role"] = [ary[4]]
+        elif ary[3] == "willshersystems" and ary[4] == "ansible-sshd":
+            ctx["role"] = ["sshd"]
+        elif ary[3] == "performancecopilot" and ary[4] == "ansible-pcp":
+            ctx["role"] = ["metrics"]
+        else:
+            raise Exception("unknown url " + ctx["github_url"])
+    if not ctx["status"]:
+        if gh_issue["is_merged"]:
+            ctx["status"] = "In Progress"
+        else:
+            ctx["status"] = "Planning"
+    if not ctx["dev_ack"]:
+        if gh_issue["is_pull"]:
+            ctx["dev_ack"] = "Dev ack"
+    if not ctx["doc_text_type"]:
+        if ctx["issue_type"] == "Story":
+            ctx["doc_text_type"] = "Enhancement"
+        elif ctx["issue_type"] == "Bug":
+            ctx["doc_text_type"] = "Bug Fix"
+    if not ctx["doc_text"]:
+        if gh_issue["is_pull"]:
+            ctx["doc_text"] = ctx["description"]
+    if not ctx["docs_impact"]:
+        if ctx["doc_text_type"]:
+            ctx["docs_impact"] = "Yes"
+        else:
+            ctx["docs_impact"] = "No"
+
+
+# these are all fields that can use the string format method
+# to construct the final value from variables derived from
+# previous issues
+TEMPLATE_FIELDS = ["description", "epic_name", "summary"]
+
+
+# this handles creating the jira issue, updating it,
+# doing the status transition, adding links, etc.
+def __create_issue(kwargs):
```

Comment:
Maybe rename this to __manage_issue if this handles updating them too?

Reply:
Right now it only handles creating new issues - note that the way that jira works, you cannot set all of the fields in the create operation - some fields, like story points, must be set after creation - that's why `__create_issue` calls `__update_issue` if there are fields which cannot be set in the create operation.